### PR TITLE
NIFI-2795 Sys Diagnostics in Cluster UI

### PR DESCRIPTION
**Pull request by jvwing**
Reworked the existing cluster UI to provide tabs containing multiple data tables.  Added views for System, JVM, FlowFile Storage and Content Storage diagnostics.  This is a UI-only change built on top of the existing System Diagnostics API.

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/cluster/nf-cluster-table.js [hunk 1]
```diff
@@ -377,190 +952,45 @@ nf.ClusterTable = (function () {
...
-            if (nf.Common.isDefinedAndNotNull(clusterGrid)) {
-                clusterGrid.resizeCanvas();
-            }
+            clusterTabs.forEach(function (tab) {
+                try {
+                    if (tab.grid) {
+                        tab.grid.resizeCanvas();
+                    }
+                } catch (ex) {
+                    console.error("Failed to resize tab", tab, ex);
```
Comment (jvwing) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/cluster/nf-cluster-table.js]: There is no good reason for it to fail, I'll change this.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/cluster/nf-cluster-table.js [hunk 2]
```diff
@@ -24,13 +24,275 @@ nf.ClusterTable = (function () {
...
+        filterOptions: [{
+            text: 'by address',
+            value: 'address'
+        }, {
+           text: 'by repository',
+           value: 'contentRepoId'
+       }]
+    };
+
+    var clusterTabs = [nodesTab, systemTab, jvmTab, flowFileTab, contentTab];
```
Comment (jvwing) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/cluster/nf-cluster-table.js]: That's a great idea, @mcgilman, I'll look into it.

**Comment by jvwing** [comment]
I added server- and client-side modifications to pre-authorize the system diagnostics API call, and removed the unnecessary error handling code.

**Comment by mcgilman** [comment]
Reviewing...

**Comment by mcgilman** [comment]
@jvwing While reviewing I stumbled into an existing bug filtering the tables in the cluster page. I've addressed it here [1]. Hoping you could just review what I did so I could include it in your PR. 

Thanks!

[1] https://github.com/mcgilman/nifi/commit/45d9fc33054a5696b67bec7a934a4e4500a01ad5

**Comment by jvwing** [comment]
@mcgilman Your changes look good to me.  I had noticed that filter behavior, but I wasn't sure how intentional it was.  Your fix does allow filtering by the full address or status text as displayed, rather than just one column value, and that seems more intuitive to users.  

Should I merge that commit into the PR branch?

**Comment by mcgilman** [comment]
@jvwing I'll just squash those changes into your commit before I push. Just wanted to communicate before I do that. Thanks!

**Comment by mcgilman** [comment]
@jvwing Just finished up testing and everything looks good. This has been merged to master. I did find another minor issue with sorting the table but I updated the sort method accordingly. Thanks!

**Comment by jvwing** [comment]
Thank you, @mcgilman.

**Pull request closed without merging**